### DISCUSSION TITLE
Receive OSC data from MA3

### DISCUSSION THREAD

**Discussion by SkowroFl** [sstaub/gma3#9] [Ideas]
Hello Stefan,

in the MA3 forum is a LUA plugin to send fader and key data via OSC. On MA3 side this code works fine, but with the receiveUDP() function in your code I have some trouble.

LUA code: [https://forum.malighting.com/forum/thread/8083-how-to-get-the-current-value-of-an-executor-fader/](url)

1. The data format is different, I do not see the information
2. Sending a lot of data from MA3 only half of them arrives

Example:
MA3 Out: SendOSC 2 "Fader203,i,1800"

Arduino In: OSC Pattern: /Fader203 String:  Integer 1: 0 Integer 2: 0 Float: 0.00

Can you help me there to modify the code?

**Comment by sstaub** [comment]
The Link doesn't work.

**Reply by SkowroFl** [reply]
Sorry, do not know why. Here again, I am working with the code of post #18.

https://forum.malighting.com/forum/thread/8083-how-to-get-the-current-value-of-an-executor-fader/?postID=20906#post20906

**Comment by sstaub** [comment]
The problem is the format of the message. The parser use the structure of the feedback, see also the documentation.
You should send following format:
/gma3, sif, fader, fader number, fader value
/gma3 // the default OSC Pattern that indicates data from GrandMA3
sii // send a string for the object type "fader" (or "key"...), an integer for the object number, and a float for the value
This structure allows a simple parsing of the necessary data.
e.g. SendOSC 2 "/gma3, sif, fader, 203, 18.00"
for fader 203 with 18% value, hope I formatted correctly
